comp/core/workloadmeta/collectors/internal/containerd/containerd.go

```diff
@@ -185,6 +189,12 @@ func (c *collector) stream(ctx context.Context) {
 	healthHandle := health.RegisterLiveness(componentName)
 	ctx, cancel := context.WithCancel(ctx)
 
+	defer func() {
+		if c.queue != nil {
+			c.queue.ShutDown()
+		}
+	}()
```

Comment:
IMO this defer() should be part of `startSBOMCollection()` since its instantiation are done  there.